staging/src/k8s.io/client-go/tools/clientcmd/client_config.go

```diff
@@ -490,9 +490,9 @@ func (config *inClusterClientConfig) ClientConfig() (*restclient.Config, error)
 		if server := config.overrides.ClusterInfo.Server; len(server) > 0 {
 			icc.Host = server
 		}
-		if token := config.overrides.AuthInfo.Token; len(token) > 0 {
-			icc.BearerToken = token
-			icc.BearerTokenFile = ""
```

Comment:
I didn't realize the overrides had a tokenfile field. Updated this to honor that field and added tests.